### PR TITLE
Feature/final login logout

### PR DESCRIPTION
**Pull request by forever-Agriculture**

### CODE REVIEW

File: backend/config.py.example [hunk 1]
```diff
@@ -22,3 +22,6 @@ class ProductionConfig(Config):
 
 class DevelopmentConfig(Config):
     DEBUG = True
+
+# password hashing rounds
+BCRYPT_LOG_ROUNDS = 14
```
Comment (serge30) [inline backend/config.py.example]: Shouldn't it be in `Config` class?

File: backend/create_database.py [hunk 2]
```diff
@@ -21,9 +21,10 @@ def db_create():
         create_database(db_credentials)
         db.create_all()
         db.session.commit()
-        print "Successfully created database and tables."
+        print("Successfully created database and tables.")
```
Comment (serge30) [inline backend/create_database.py]: `print` in Python 2.x is not function, but statement. Please remove parentheses.

File: backend/models/users.py [hunk 3]
```diff
@@ -12,18 +12,29 @@ class Role(db.Model):
...
+    """User table in the database"""
 
     __tablename__ = 'users'
 
     id = db.Column(db.Integer, primary_key=True)
     name = db.Column(db.Text)
     alias = db.Column(db.Text)
     email = db.Column(db.Text)
-    password = db.Column(db.Text)
+    _password = db.Column(db.Text, nullable=False)
```
Comment (serge30) [inline backend/models/users.py]: I'd rather do not change field name. It's not common to use underscore prefixes for DB column names.

File: requirements.txt [hunk 4]
```diff
@@ -1,4 +1,5 @@
 alembic==0.9.6
+bcrypt==3.1.4
```
Comment (serge30) [inline requirements.txt:2]: You use `Flask-Bcrypt` module, but add only `bcrypt` as dependency.

File: backend/views/views.py [hunk 5]
```diff
@@ -105,22 +106,22 @@ def login():
     if form.validate_on_submit():
         user = db.session.query(User).filter(
             User.email == form.email.data).first()
-        if user and user.password == form.password.data:
+        if user.is_correct_password(form.password.data):
             session['user_id'] = user.id
             session['role_id'] = user.role_id
-            flash('Wellcome %s' % user.name)
+            flash('Welcome {}'.format(user.name))
```
Comment (serge30) [inline backend/views/views.py]: I'm not sure you need to change formating method here.

**Review by serge30 — CHANGES_REQUESTED** [review]
Please set your dev environment to do not change file permissions mode, as in this PR you triggered all files. Please revert all files mode to standard one (0644).

File: backend/config.py.example (hunk 1 above)
Comment (forever-Agriculture) [reply backend/config.py.example]: shame on me, fixed

File: backend/create_database.py (hunk 2 above)
Comment (forever-Agriculture) [reply backend/create_database.py]: +

File: requirements.txt (hunk 4 above)
Comment (forever-Agriculture) [reply requirements.txt:2]: Flask-Bcrypt added

File: backend/models/users.py (hunk 3 above)
Comment (forever-Agriculture) [reply backend/models/users.py]: the idea was to notify coders that passwords will be stored with hashing (not in a raw state). I've seen such things in examples, so coders are aware that some additional stuff will happen with passwords. Just considered that was a nice approach. But if it's required i can change the name

File: backend/views/views.py (hunk 5 above)
Comment (forever-Agriculture) [reply backend/views/views.py]: fine :(

File: requirements.txt [hunk 6]
```diff
@@ -11,7 +14,9 @@ itsdangerous==0.24
 Jinja2==2.9.6
 Mako==1.0.7
 MarkupSafe==1.0
+pep8==1.7.1
```
Comment (serge30) [inline requirements.txt]: I do not think, `pep8` is required to run this application :)

File: backend/models/users.py [hunk 7]
```diff
@@ -12,18 +12,29 @@ class Role(db.Model):
...
+    @hybrid_property
+    def password(self):
+        return self._password
+
+    # hashing password before being stored
+    @password.setter
+    def _set_password(self, plaintext):
+        self._password = bcrypt.generate_password_hash(plaintext)
+
+    def is_correct_password(self, plaintext):
```
Comment (serge30) [inline backend/models/users.py]: `check_password` sounds a bit better.

File: backend/models/users.py (hunk 3 above)
Comment (serge30) [reply backend/models/users.py]: I'm ok to have object's attribute `_password`, but field with the same name will be created in DB, what is not good.

**Review by serge30 — CHANGES_REQUESTED** [review]

File: backend/models/users.py (hunk 7 above)
Comment (forever-Agriculture) [reply backend/models/users.py]: absolutely agree

File: backend/models/users.py (hunk 3 above)
Comment (forever-Agriculture) [reply backend/models/users.py]: consensus

File: backend/insert_db_data.py [hunk 8]
```diff
@@ -53,17 +53,17 @@
...
-                        issue1, issue2, issue3,
-                        issueHistory1, issueHistory2, issueHistory3,
-                        issueHistory4, issueHistory5
+    db.session.add_all([#role, role1, role2,
+                        #category, category1, category2,
+                        #status1, status2, status3,
+                        user1, user2, user3
+                        #issue1, issue2, issue3,
+                        #issueHistory1, issueHistory2, issueHistory3,
+                        #issueHistory4, issueHistory5
```
Comment (serge30) [inline backend/insert_db_data.py]: What is the reasons of those changes?

Comment (forever-Agriculture) [reply backend/insert_db_data.py]: fixed, just have been focused on users, sorry

**Review by serge30 — APPROVED** [review]
Approved.

**Pull request merged**